data_structures/linked_list/reverse_linked_list_program.py

```diff
@@ -0,0 +1,55 @@
+# Time Complexity : O(n) 
+# Space Complexity : O(1) 
+
+# Node class 
+class Node: 
+
+	# Constructor to initialize the node object 
+	def __init__(self, data): 
+		self.data = data 
+		self.next = None
+
+class LinkedList: 
+
+	# Function to initialize head 
+	def __init__(self): 
+		self.head = None
+
+	# Function to reverse the linked list 
+	def reverse(self): 
+		prev = None
+		current = self.head 
+		while(current is not None): 
+			next = current.next
+			current.next = prev 
+			prev = current 
+			current = next
+		self.head = prev 
+		
+	# Function to insert a new node at the beginning 
+	def push(self, new_data): 
+		new_node = Node(new_data) 
+		new_node.next = self.head 
+		self.head = new_node 
+
+	# Utility function to print the linked LinkedList 
+	def printList(self): 
+		temp = self.head 
+		while(temp): 
+			print(temp.data) 
+			temp = temp.next
+
+
+# Driver program to test above functions 
+llist = LinkedList() 
+llist.push(20) 
+llist.push(4) 
+llist.push(15) 
+llist.push(85) 
+
+print("Given Linked List")
+llist.printList() 
+llist.reverse() 
+print("\nReversed Linked List")
+llist.printList() 
+
```

Comment:
This should be under `if __name__ == "__main__":` block

Also, your entire code is in 8 spaces while Python uses only 4 spaces as indentation.